src/test/expression/expression_evaluator_test.cpp

```diff
@@ -482,6 +487,45 @@ TEST_F(ExpressionEvaluatorTest, InSelectUncorrelated) {
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(34.5, select_b), {std::nullopt}));
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_("hello", select_b), {0}));
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(c, select_b), {1, std::nullopt, 1, std::nullopt}));
+
+  // Test it with pre-calculated uncorrelated_select_results
+  auto uncorrelated_select_results = std::make_shared<ExpressionEvaluator::UncorrelatedSelectResults>();
+  table_wrapper_a->execute();
+  pqp_a->execute();
+  table_wrapper_b->execute();
+  pqp_b->execute();
+  uncorrelated_select_results->emplace(pqp_a, pqp_a->get_output());
+  uncorrelated_select_results->emplace(pqp_b, pqp_b->get_output());
+
+  // Destroy the pqp's so that if the expression evaluator tries to access them, it crashes
+  pqp_a->input_left() = nullptr;
+  pqp_b->input_left() = nullptr;
+
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(6, select_a), {0}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(a, select_a), {1, 1, 1, 1}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(add_(a, 2), select_a), {1, 1, 0, 0}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(b, select_a), {1, 1, 1, 0}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(34, select_b), {1}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(34.0, select_b), {1}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(34.5, select_b), {std::nullopt}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_("hello", select_b), {0}, uncorrelated_select_results));
+  EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(c, select_b), {1, std::nullopt, 1, std::nullopt},
+                                       uncorrelated_select_results));
+
+  // Make sure the expression evaluator complains if it has been given a list of preevaluated selects but one is missing
```

Comment:
Minor (and I get you just want to reuse the pqps from above): The fact that you have to add these structuring comments suggests that these should be three distinct test cases.

Reply:
done